frontend/apps/reader/modules/readersearch.lua

```diff
@@ -145,6 +146,17 @@ function ReaderSearch:onShowSearchDialog(text, direction)
                 -- Don't add result pages to location ("Go back") stack
                 neglect_current_location = true
             end
+            if res==nil or (current_page~=nil and valid_link==nil) then
+                local notification_text
+                if self._expect_back_results then
+                    notification_text = _("No results backward")
+                else
+                    notification_text = _("No results forward")
```

Comment:
(Or `No previous result` / `No next result` ?)

Reply:
> (Or `No previous result` / `No next result` ?)

Is it okay for initial search with nothng found?

Reply:
OK, well, I dunno :) English scholars to the rescue!

Reply:
Oh, yes, this doesn't work ^^.

New question: is the distinction really necessary? I'm inclined to think a simple "No results" works just as well, and will in fact be less confusing.

Reply:
Imagine you are on the last page with the results highlighted and press `▷`.
Is just `No results` good for that?

Also for initial search. `No results` in the entire book or in the next pages only?

Reply:
Well, yes, no result for the action the user just requested makes sense to me, you're not supposed to have forgotten which button you've tapped on 10ms ago?

Reply:
Well, I might hit `>` without thinking much that I'm searching forward only.
`No result` won't help me realize I could find results backwards.
As we know that fact, it feels a bit sad to not let it known one way or another (although I probably won't read a 2s notification text - but the longer the text, the more I'll be curious to focus on it :)

Reply:
That raises another question, though: Assuming "No results ahead" / "No results behind": ahead/behind of *what*?

Reply:
No results or Not found
... on previous pages / ... on next pages
... back / ... ahead
... backward / ... forward (onward)
... before / ... after

Reply:
IIRC, "page" isn't really what it's actually relative to, but at least it's something, and something that's mostly accurate in practice, so, yeah, that works.

previous / *following* pages, though ;).

Reply:
I'd be inclined to say preceding / following.